core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -317,6 +327,11 @@ public void readFromThin(StreamInput in) throws IOException {
         if (in.readBoolean()) {
             allocationId = new AllocationId(in);
         }
+        if (relocating() || initializing()) {
+            expectedShardSize = in.readLong();
```

Comment:
I assume 2.0.0.beta1 will not be wire-compatible with 2.0.0 GA?  So we don't need any version checks when we serialize/deserialize here...